File: lib/models/pet_model.dart
class Pet {
  late String nome;
  late String imageUrl;
  late String descricao;
  late int idade;
  late String sexo;
  late String cor;
  late String bio;
  late int id;

  Pet({
  required this.nome,
  required this.imageUrl,
  required this.descricao,
  required this.idade,
  required this.sexo,
  required this.cor,
  required this.bio,
  required this.id
  });

  Map<String, Object> toMap() {
    return {
      'id': id,
      'nome': nome,
      'imageUrl': 'assets/images/toto.png',
      'descricao': descricao,
      'idade': idade,
      'sexo': sexo,
      'cor': cor,
      'bio': bio
    };
  }

  Pet.fromMap(Map map) {
    id= map['id'];
    nome= map['nome'];
    imageUrl= map['imageUrl'];
    descricao= map['descricao'];
    idade= map['idade'];
    cor= map['cor'];
    sexo= map['sexo'];
    bio= map['bio'];
  }
}

File: lib/models/pet_model.php
Sclass Pet {
public $nome;
public $imageUrl;
public $descricao;
public $idade;
public $sexo;
public $cor;
public $bio;
public $id;

function __construct($nome, $imageUrl, $descricao, $idade, $sexo, $cor, $bio, $id) {
$this->nome = $nome;
$this->imageUrl = $imageUrl;
$this->descricao = $descricao;
$this->idade = $idade;
$this->sexo = $sexo;
$this->cor = $cor;
$this->bio = $bio;
$this->id = $id;
}

public function toMap() {
return [
'id' => $this->id,
'nome' => $this->nome,
'imageUrl' => 'assets/images/toto.png',
'descricao' => $this->descricao,
'idade' => $this->idade,
'sexo' => $this->sexo,
'cor' => $this->cor,
'bio' => $this->bio
];
}

public static function fromMap($map) {
return new Pet($map['nome'], $map['imageUrl'], $map['descricao'], $map['idade'], $map['sexo'], $map['cor'], $map['bio'], $map['id']);
}
}